app/src/test/kotlin/com/intive/picover/common/result/ImageFileRepositoryTest.kt

```diff
@@ -1,43 +1,56 @@
-package com.intive.picover.camera.repository
+package com.intive.picover.common.result
 
+import android.app.Activity
 import android.content.Context
+import android.content.Intent
 import android.net.Uri
 import android.os.Environment
+import androidx.activity.result.contract.ActivityResultContracts
 import androidx.core.content.FileProvider
 import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.datatest.withData
 import io.kotest.matchers.shouldBe
 import io.mockk.every
 import io.mockk.mockk
+import io.mockk.mockkConstructor
 import io.mockk.mockkStatic
 import io.mockk.unmockkAll
 import java.io.File
 
 class ImageFileRepositoryTest : ShouldSpec(
 	{
 		val context: Context = mockk()
-		val tested = ImageFileRepository(context)
+		val tested = TakePictureContract()
 
 		beforeSpec {
 			mockkStatic(File::class, FileProvider::class)
+			mockkConstructor(ActivityResultContracts.TakePicture::class)
 		}
 
 		afterSpec {
 			unmockkAll()
 		}
 
-		should("return image file uri WHEN createTempFileAndGetUri") {
+		context("parse result WHEN created temp file uri passed to take picture delegate") {
 			val packageName = "com.intive.picover"
 			val externalFilesDir: File = mockk()
 			val tempFile: File = mockk()
-			val imageUri: Uri = mockk()
+			val pictureUri: Uri = mockk()
+			val intent: Intent = mockk()
 			every { context.getExternalFilesDir(Environment.DIRECTORY_PICTURES) } returns externalFilesDir
 			every { context.packageName } returns packageName
 			every { File.createTempFile("IMG_", ".jpg", externalFilesDir) } returns tempFile
-			every { FileProvider.getUriForFile(context, "$packageName.provider", tempFile) } returns imageUri
+			every { FileProvider.getUriForFile(context, "$packageName.provider", tempFile) } returns pictureUri
```

Comment:
For future tests we can relax testing approach in cases like this one. We can `any()` instead of exact parameters like `"IMG_"`, ".jpg", etc. because in the end, the test is almost a copy of the implementation, what needs to be tested manually anyway, by launching the app. Such I/O operations should be tested in scope of the instrumentation tests.